pandas/core/arrays/period.py

```diff
@@ -341,9 +341,7 @@ def freq(self) -> BaseOffset:
     def __array__(self, dtype: NpDtype | None = None) -> np.ndarray:
         if dtype == "i8":
             return self.asi8
-        # error: Non-overlapping equality check (left operand type: "Optional[Union[str,
-        # dtype[Any]]]", right operand type: "Type[bool]")
-        elif dtype == bool:  # type: ignore[comparison-overlap]
```

Comment:
This is a false positive from numpy. no need to change code. probably better to fix upstream. not found an issue on the numpy issue tracker, may need to create one.

https://github.com/numpy/numpy/blob/9b192570661d55575606f2a77a7bc9e280120f5c/numpy/__init__.pyi#L1072-L1075

AFAICT without definitions for \_\_eq__ and \_\_ne__, these are inheriting from object which doesn't have Type[something] in the annotations

Reply:
OK, most of the edits in this PR are similar, so can probably close

Reply:
it would have been quicker for me to add quotes to all these rather than adding the comment, ignore and reformatting but I tend to try to avoid code changes for false positives in most cases. 

I'm not against the code changes if you think it's an improvement from an aspect other than the mypy ignores (which if gets fixed upstream can be removed)